### PR TITLE
Update updater.sh

### LINKED ISSUE #120: Update software update helper bash script
Currently the script just updates the remote app itself:
```
#!/bin/bash
fbv -d 1 /update.png
killall -9 python
killall -9 remote
sleep 2
cp /usr/bin/remote /usr/bin/remote-backup
rm -rf /usr/bin/remote
sleep 2
curl -L $1 > /usr/bin/remote
sleep 2
chmod +x /usr/bin/remote
/usr/bin/app-launch.sh &
```

It should be updated to pull a zip file, that contains all the helper scripts and config files, etc and replace everything that is inside /usr/bin/remote.

1. kill python and remote
2. make a backup of the /usr/bin/remote folder
3. delete the folder
4. download the URL that is given to the bash script > $1
4. extract the zip file to /usr/bin/remote
5. give executable permissions to all .sh files and the remote file
6. launch the remote app with the launcher bash script

### PR DESCRIPTION
**Pull request by gvdhoven**
Should fix #120

### CODE REVIEW

**Comment by carp3-noctem** [comment]
As i think this will be run on the pi to update the remote to the newest OS, i don't think the root is required, as you are free to choose where to store the files and the pi itself isn't locked.
Am i right @martonborzak ?
Also it may be a good idea to stow the Backup made by the OS and this one in the same location and provide the Timestamp (as mentioned) so the user is able to choose which backup to use.

Lets say:
- Keep maximum of 20 Backups (one user Favorite that doesn't get killed)
- Make a new with each change to config (without touching the Favorite)
- Make a new with each Software Backup (without touching the Favorite)

**Comment by martonborzak** [comment]
@carp3-noctem You're right. This updater will only run on the pi. I think it doesn't make sense to run it on the development machines.

The pi has an OS built with buildroot. Therefore it's not possible to update packages there, just the application and helper scripts. Because of buildroot, you get a default root user with a predefined password.

Good idea to keep multiple backups with timestamps. Those could be also downloaded via the web interface proposed in #97

File: software/helper scripts/updater.sh [hunk 1]
```diff
@@ -1,12 +1,85 @@
 #!/bin/bash
-fbv -d 1 /update.png
+
+#--------------------
+# YIO updater script
+#--------------------
+if [[ $EUID -ne 0 ]]; then
```
Comment (martonborzak) [inline software/helper scripts/updater.sh]: I think we don't need this as everything runs as root on the pi.

File: software/helper scripts/updater.sh [hunk 2]
```diff
@@ -1,12 +1,85 @@
...
+#
+# Since the developers of the yio-remote use macs, i adopted this script to work with the /usr/local/yio-remote target path.
+# That is also why we require root access
+#--------------------
+
+
+#--------------------
+# TODO: Rework based on path?
+#--------------------
+fbv -d 1 /usr/local/yio-remote/update.png
```
Comment (martonborzak) [inline software/helper scripts/updater.sh]: Same as above, if we only run it on the pi

File: software/helper scripts/updater.sh (hunk 1 above)
Comment (martonborzak) [inline software/helper scripts/updater.sh]:
```suggestion
```

File: software/helper scripts/updater.sh [hunk 3]
```diff
@@ -1,12 +1,85 @@
 #!/bin/bash
-fbv -d 1 /update.png
+
+#--------------------
+# YIO updater script
+#--------------------
+if [[ $EUID -ne 0 ]]; then
+   echo "This script must be run as root (probably not a good idea to run this script yourself)." 1>&2
```
Comment (martonborzak) [inline software/helper scripts/updater.sh]:
```suggestion
```

File: software/helper scripts/updater.sh [hunk 4]
```diff
@@ -1,12 +1,85 @@
...
+   exit 1
```
Comment (martonborzak) [inline software/helper scripts/updater.sh]:
```suggestion
```

File: software/helper scripts/updater.sh [hunk 5]
```diff
@@ -1,12 +1,85 @@
...
+fi
```
Comment (martonborzak) [inline software/helper scripts/updater.sh]:
```suggestion
```

File: software/helper scripts/updater.sh (hunk 2 above)
Comment (martonborzak) [inline software/helper scripts/updater.sh]:
```suggestion
fbv -d 1 /usr/bin/yio-remote/update.png
```

File: software/helper scripts/updater.sh [hunk 6]
```diff
@@ -1,12 +1,85 @@
...
+# TODO: Rework based on path?
+#--------------------
+fbv -d 1 /usr/local/yio-remote/update.png
+
+
+#--------------------
+# 1. Create temp location
+#    - Download file and pipe to tar to unzip to temp location
+#--------------------
+tmpdir="/usr/local/yio-$(date +"%Y%m%d%H%M%S")"
```
Comment (martonborzak) [inline software/helper scripts/updater.sh]:
```suggestion
tmpdir="/usr/bin/yio-$(date +"%Y%m%d%H%M%S")"
```

File: software/helper scripts/updater.sh [hunk 7]
```diff
@@ -1,12 +1,85 @@
...
+# 3. Give executable permissions to all .sh files
+#--------------------
+find "${tmpdir}" -type f -name "*.sh" -exec chmod 775 {} +
+chmod +x ${tmpdir}/remote
+
+
+#--------------------
+# 4. Remove previous backups (should not be needed, unless this scripts fails somewhere, therefore not yet implemented)
+#--------------------
+rm -rf "/usr/local/yio-remote-backup"
```
Comment (martonborzak) [inline software/helper scripts/updater.sh]:
``` suggestion
rm -rf "/usr/bin/yio-remote-backup"
```

File: software/helper scripts/updater.sh [hunk 8]
```diff
@@ -1,12 +1,85 @@
...
+
+
+#--------------------
+# 5. Make a backup of the /usr/bin/remote folder (if not already exist, maybe add timestamp to folder)
+#--------------------
+mv "/usr/local/yio-remote" "/usr/local/yio-remote-backup"
```
Comment (martonborzak) [inline software/helper scripts/updater.sh]:
``` suggestion
mv "/usr/bin/yio-remote" "/usr/bin/yio-remote-backup"
```

File: software/helper scripts/updater.sh [hunk 9]
```diff
@@ -1,12 +1,85 @@
...
+
+
+#--------------------
+# 6. Kill python and remote by pid
+#    - todo/improvement: Only kill based on PiD to prevent killing other apps?
```
Comment (martonborzak) [inline software/helper scripts/updater.sh:53]: I think we can kill apps like this, as nothing else will run on the pi

File: software/helper scripts/updater.sh [hunk 10]
```diff
@@ -1,12 +1,85 @@
...
-sleep 2
-chmod +x /usr/bin/remote
-/usr/bin/app-launch.sh &
\ No newline at end of file
+
+
+#--------------------
+# 8. Rename the update folder to /usr/local/yio-remote
+#--------------------
+mv "${tmpdir}" "/usr/local/yio-remote"
```
Comment (martonborzak) [inline software/helper scripts/updater.sh]:
``` suggestion
mv "${tmpdir}" "/usr/bin/yio-remote"
```

File: software/helper scripts/updater.sh [hunk 11]
```diff
@@ -1,12 +1,85 @@
...
+
+
+#--------------------
+# 9. Launch the remote app with the launcher bash script
+#--------------------
+/usr/local/yio-remote/app-launch.sh &
```
Comment (martonborzak) [inline software/helper scripts/updater.sh]:
``` suggestion
/usr/bin/yio-remote/app-launch.sh & 
```

**Review by martonborzak — COMMENTED** [review]
Thank you for looking at it @Webunity!

I think it's cool. I made suggestions to change the directory. If you agree, let's merge it and I'll test it.

File: software/helper scripts/updater.sh (hunk 2 above)
Comment (gvdhoven) [reply software/helper scripts/updater.sh]: yeah but currently it did `fbv -d 1 /update.png` (it was missing a path); can it be then this: `fbv -d 1 /usr/bin/yio-remote/update.png` ?

Comment (gvdhoven) [reply software/helper scripts/updater.sh]: edit: didn't see your change below; this is resolved.

**Pull request merged**